File: NQT/Numbers/Progressions/ArthmeticProgressionSum.java
package Numbers.Progressions;

public class ArthmeticProgressionSum {
    public static void main(String[] args) {
        int firstTerm = 1; // First term of the arithmetic progression
        int commonDifference = 2; // Common difference
        int numberOfTerms = 5; // Number of terms to sum

        int sum = calculateAPSum(firstTerm, commonDifference, numberOfTerms);
        int sumUsingLoop = calculateAPSumUsingLoop(firstTerm, commonDifference, numberOfTerms);
        System.out.println("The sum of the first " + numberOfTerms + " terms of the arithmetic progression is: " + sum);
        System.out.println("The sum of the first " + numberOfTerms + " terms of the arithmetic progression (using loop) is: " + sumUsingLoop); 
    }

    private static int calculateAPSum(int firstTerm, int commonDifference, int numberOfTerms) {
        // Using the formula: S_n = n/2 * (2a + (n-1)d)
        return (numberOfTerms / 2) * (2 * firstTerm + (numberOfTerms - 1) * commonDifference);
    }

    // Alternatively, you can calculate the sum using a loop:
    private static int calculateAPSumUsingLoop(int firstTerm, int commonDifference, int numberOfTerms) {
        int sum = 0;
        for (int i = 0; i < numberOfTerms; i++) {
            sum += firstTerm + i * commonDifference;
            System.out.print("Term " + (i + 1) + ": " + (firstTerm + i * commonDifference)); // Print each term
        }
        return sum;
    }
}

File: NQT/Numbers/Progressions/arthmetic_progression_sum.py
# find the sum of the first n terms of an arithmetic progression

# Using the formula for the sum of an arithmetic progression
def arithmetic_progression_sum(first_term, common_difference, number_of_terms):
    """
    Calculate the sum of the first n terms of an arithmetic progression.

    Parameters:
    first_term (float): The first term of the arithmetic progression.
    common_difference (float): The common difference between consecutive terms.
    number_of_terms (int): The number of terms to sum.

    Returns:
    float: The sum of the first n terms of the arithmetic progression.
    """
    sum_n = number_of_terms / 2 * (2 * first_term + (number_of_terms - 1) * common_difference)
    return sum_n

# Using the for loop to calculate the sum of the first n terms of an arithmetic progression
def arithmetic_progression_sum_loop(first_term, common_difference, number_of_terms):
    """
    Calculate the sum of the first n terms of an arithmetic progression using a loop.

    Parameters:
    first_term (float): The first term of the arithmetic progression.
    common_difference (float): The common difference between consecutive terms.
    number_of_terms (int): The number of terms to sum.

    Returns:
    float: The sum of the first n terms of the arithmetic progression.
    """
    sum_n = 0
    for i in range(number_of_terms):
        sum_n += first_term + i * common_difference
    return sum_n

# Example usage
if __name__ == "__main__":
    first_term = 2
    common_difference = 3
    number_of_terms = 5

    sum_formula = arithmetic_progression_sum(first_term, common_difference, number_of_terms)
    sum_loop = arithmetic_progression_sum_loop(first_term, common_difference, number_of_terms)

    print(f"Sum of the first {number_of_terms} terms (using formula): {sum_formula}")
    print(f"Sum of the first {number_of_terms} terms (using loop): {sum_loop}")